hypershift-operator/controllers/nodepool/aws.go

```diff
@@ -63,6 +64,15 @@ func awsMachineTemplateSpec(infraName, ami string, hostedCluster *hyperv1.Hosted
 			Filters: filters,
 		})
 	}
+	if len(securityGroups) == 0 && defaultSG {
+		if hostedCluster.Status.Platform == nil || hostedCluster.Status.Platform.AWS == nil || hostedCluster.Status.Platform.AWS.DefaultWorkerSecurityGroupID == "" {
+			return nil, &NotReadyError{fmt.Errorf("the default security group for the HostedCluster has not been created")}
+		}
+		sgID := hostedCluster.Status.Platform.AWS.DefaultWorkerSecurityGroupID
+		securityGroups = append(securityGroups, capiaws.AWSResourceReference{
```

Comment:
fwiw, this will trigger a rolling upgrade for all existing machineDeployments (ocm) - which is fine I guess at this stage.

Reply:
Because we only do this when we're using a control plane operator that has this capability (creates a default sg), you'd need to upgrade existing HC's for this to take effect